text_generation/causal_lm/cpp/continuous_batching/library/src/continuous_batching_pipeline.cpp

```diff
@@ -147,6 +160,24 @@ class ContinuousBatchingPipeline::Impl {
             timer.end();
         }
 
+        // if no tokens were scheduled, we are out of memory
+        if (scheduler_output.m_total_num_scheduled_tokens == 0) {
+            for (size_t sequence_group_id = 0; sequence_group_id < m_requests.size(); ++sequence_group_id) {
+                m_requests[sequence_group_id]->set_out_of_memory();
```

Comment:
do you think we can move this loop to scheduler? In this case, we will simplify this high-level code, which only post-process the scheduler results (setting out_of_memory flag is also a scheduler result, IMO)

Reply:
Yes, moved this cycle to `schedule()` method.